fhir-persistence/src/main/java/com/ibm/fhir/persistence/util/FHIRPersistenceUtil.java

```diff
@@ -210,6 +212,50 @@ public static Resource createDeletedResourceMarker(Resource deletedResource) {
         }
     }
 
+    public static Resource createDeletedResourceMarker(String resourceType, String logicalId, int version, java.time.Instant lastUpdated) {
+        // TODO do we even need this deletion marker now that we have ResourceResult?
+        try {
+            // Build a fresh meta with only versionid/lastupdated defined
+            Meta meta = Meta.builder()
+                    .versionId(Id.of(Integer.toString(version)))
+                    .lastUpdated(Instant.of(lastUpdated.atZone(ZoneOffset.UTC)))
+                    .build();
+
+            // Build a minimal instance of Resource
+            Class<? extends Resource> resourceClass = ModelSupport.getResourceType(resourceType);
+            Resource resource = resourceClass.getDeclaredConstructor().newInstance()
```

Comment:
How does this work when our resource constructors all require a Builder?!

Anyway, seeing this makes me feel even stronger that its probably worth trying to remove any need for the resource marker rather than trying to construct the marker instances out of thin air.

Reply:
Right, I remember this was getting horrible so I'm glad we don't have to use it now. Cleaning it out.